test/integration/command-logging-and-monitoring/command_logging_and_monitoring.prose.test.ts

```diff
@@ -0,0 +1,238 @@
+          buffer: [],
+          write(log) {
+            this.buffer.push(log);
+          }
+        };
+
+        // 1.
+        const client = this.configuration.newClient(
+          {},
+          {
+            [loggerFeatureFlag]: true,
+            mongodbLogPath: writable,
+            mongodbLogComponentSeverities: {
+              command: 'debug'
+            },
+            mongodbLogMaxDocumentLength: 5
+          }
+        );
+
+        // 2.
+        await client.db('admin').command({ hello: 'true' });
+
+        // remove sensitive commands for uniformity
+        const cleanedBuffer = [];
+        for (let i = 0; i < writable.buffer.length; i++) {
+          if (writable.buffer[i].command !== '{}' && writable.buffer[i].reply !== '{}') {
+            cleanedBuffer.push(writable.buffer[i]);
+          }
+        }
+
+        // 3.
+        const insertManyCommandStarted = cleanedBuffer[0];
+        expect(insertManyCommandStarted?.message).to.equal('Command started');
+        expect(insertManyCommandStarted?.command).to.be.a('string');
+        expect(insertManyCommandStarted?.command?.length).to.equal(5 + ELLIPSES_LENGTH);
+
+        // 4.
+        const insertManyCommandSucceeded = cleanedBuffer[1];
+        expect(insertManyCommandSucceeded?.message).to.equal('Command succeeded');
+        expect(insertManyCommandSucceeded?.reply).to.be.a('string');
+        expect(insertManyCommandSucceeded?.reply?.length).to.be.at.most(5 + ELLIPSES_LENGTH);
+
+        await client.close();
+      }
+    });
+  });
+  context('Truncation with multi-byte codepoints', function () {
+    /*
+    A specific test case is not provided here due to the allowed variations in truncation logic
+     as well as varying extended JSON whitespace usage. 
+     Drivers MUST write language-specific tests that confirm truncation of commands, replies, 
+     and (if applicable) server responses included in error messages 
+     work as expected when the data being truncated includes multi-byte Unicode codepoints.
+     
+     If the driver uses anything other than Unicode codepoints as the unit for max document length, 
+     there also MUST be tests confirming that cases
+     where the max length falls in the middle of a multi-byte codepoint are handled gracefully.
+    */
+    it('should handle unicode codepoints in middle and end of truncation gracefully', {
+      metadata: {},
+      test: async function () {
+        const writable = {
+          buffer: [],
+          write(log) {
+            this.buffer.push(log);
+          }
+        };
+
+        const client = this.configuration.newClient(
+          {},
+          {
+            [loggerFeatureFlag]: true,
+            mongodbLogPath: writable,
+            mongodbLogComponentSeverities: {
+              command: 'debug'
+            },
+            mongodbLogMaxDocumentLength: 50
+          }
+        );
+
+        const docs: Array<Document> = [{ x: '&&&&&&&&&&&&&&&&' }];
```

Comment:
Ampersands are single byte unicode code points and won't ever be possible to split incorrectly. 

> where the max length falls in the middle of a multi-byte codepoint are handled gracefully.

This is not obvious in javascript, at least with the APIs we have, let's discuss on slack.

Reply:
Paper trail: changed ampersand to snowman emoji!

Reply:
The new test still does not prove the the point we are trying to assert. We still use .slice which will leave a string with a dangling surrogate. Snowman was an unlucky choice 😅 it is a very old emoji. Try: `'\uD801\uDC37'`